### ISSUE TITLE
Training the proximal discriminator

### ISSUE THREAD

**Issue by KranzDD** [proximal-dg/proximal_dg#3]
Hi there,

I'm trying to implement your proximal duality gap measure inside TensorFlow, but I'm having a rough time understanding your repo, due to the general nature and my PyTorch inexperience.

Before I go any deeper, do you maybe have a stand alone example of the proximal duality gap laying around? Without any general losses or architecture?

If not, here's my current question:
The algorithm in the paper defines a prox_opt function. From what I've gathered so far, this equates to the train_ops function inside the ProximalGeneratorLoss class.
This function uses the current discriminator, copies it into a new proximal discriminator and trains the proximal discriminator (using the original discriminator as part of the loss calculation).
However, as far as I can see, you are training the original discriminator:

                dx      = discriminator(real_inputs)
                dx_prox = prox_discriminator(real_inputs)
                dgz = discriminator(fake.detach())
                grad_dx      = autograd.grad(torch.unbind(dx), real_inputs, create_graph=False,retain_graph=True)[0]
                grad_dx_prox = autograd.grad(torch.unbind(dx_prox), real_inputs, create_graph=False,retain_graph=True)[0]
                penalty      = torch.mean(torch.square(torch.norm(grad_dx - grad_dx_prox,dim=(2,3))))
                loss         = self.proximal_discriminator_loss.forward(dx, dgz) + self.lamda_prox*penalty
                
                loss.backward()    
                optimizer_discriminator.step()

I have no PyTorch experience, so maybe there is something I'm missing, but it looks to me you are not training the prox_discriminator, since you are not even calcuating the loss using the prox_discriminators output.

Also, what exactly is the use of the ProximalDiscriminatorLoss class? Is it needed to calculate the proximal duality gap?

Any input is highly appreciated!
